### ISSUE TITLE
Allow for multiple ships to be targeted at once.

### ISSUE THREAD

**Issue by mdsmestad** [endless-sky/endless-sky#993]
Could be a type of systems upgrade. "Advanced Targeting" or some such thing could give you the ability to focus on multiple targets at once if you have compatible weapons.

**Comment by Amazinite** [comment]
Define "compatible weapons." Would weapons need to be made that can target multiple enemies at once? Or would it be something like having your turrets target multiple ships?

**Comment by mdsmestad** [comment]
@Amazinite Well human lazer's can't focus on two targets at once..but advanced alien weapons might be able to focus down 3 dudes at a time.

**Comment by Amazinite** [comment]
Maybe this could be used as a fleet command to have your fleet split up into two groups to focus down different targets.

**Comment by jafdy** [comment]
Or they could kill one three times faster.

**Comment by Amazinite** [comment]
Yeah, I'm not really seeing how this could work out having weapon be able to target multiple enemies. You would be just as successful at killing a group of enemies by focusing them one at a time. Maybe allowing your fleet to focus down multiple enemies at once would be useful, but I don't think having specific weapons for targeting 2+ ships would be very useful, especially because you can already hit multiple ships with submunitions or blast radius.

**Comment by lehgogh** [comment]
If you have two or more turrets, you might want them to fire at multiple targets (take down several fighters at once, rather than one at a time).

**Comment by jafdy** [comment]
Maybe if turrets take time to turn but as turrets can turn instantly unless each turret can overkill a fighter inside 1 frame then you will still take them down in the same time (maybe 1 frames longer overall).

**Comment by Musaran2** [comment]
Would be goof for :
- Tracking weapons automatically picking a target in range, instead of something they can't reach.
- Tracking weapons automatically picking the easiest target to kill.
- Tracking weapons moving on to next kill with no delay.
- Tracking weapons whose effect works best spread out.
- Incomings that can (and should) be repelled with little firepower, but hard or pointless to destroy.

**Comment by tehhowch** [comment]
Re: OP Set your turrets to `opportunistic` and they will engage whichever hostiles make the most sense, regardless if they are the true ship-based target.

Setting multiple targets on a ship-based level would require some new targeting interface - how else will you indicate this to the player, or if you choose "just put the target pointers on all of them", how do you choose the one that is displayed on the left side. You are 100% capable of designing this new interface and submitting it to the community.

Re: @Musaran2 
1. No homing weapon fires at something it can't reach. If the target decides to move in a manner that would dodge the projectile after the projectile is fired, how do you expect the firing ship to know this in advance?
2. Weakened targets are already prioritized over full-strength targets
3. Targeting (at the ship level, i.e. what ship will this ship try to point its fixed emplacements at) is checked twice per second or immediately upon destruction of the current target
4. ? 
5. ?

4 sounds like you are asking for the AI to use weapons that are best used by the player in a more effective manner (e.g. do not fire 2 Nuclear Missiles at the same target when there are two groups of hostiles in different directions). Splash damage weapons are something that is best wielded by the player, and as indicated in many MZ posts, updates to the AI that are generally useful only for large fleets are extremely low priority. Hostile AI uses them well enough (they target the player with them, which is sufficiently deadly that many players have a lot of trouble with the relevant FW:R missions), and the player should be able to make intelligent decisions already.
5... I'm not sure what you are hinting at. Saying something should get destroyed but then saying it is pointless to destroy it is not a logical statement.

**Comment by Pointedstick** [comment]
As @tehhowch mentions, `opportunistic` turrets can now do this automatically.

The issue becomes much more challenging for targeted missile weapons though, because designating targets would need to be done manually, with some kind of per-weapon or even per-shot interface, which would need both keyboard and mouse input. All of that would be possible if combat in Endless Sky wasn't so fast that by the time you got done setting your targets, they weren't already dead!

Bottom line: I don't see any feasible way that this could be done, or if it could, for it to actually work the with style of play that Endless Sky involves.
